File: iosApp/iosApp/Screens/Quotes/QuotesScreen.swift
import SwiftUI
import SharedLogic

/// S14 — one quote a day, from the bundled offline database. Paging through the
/// archive and favouriting are P1 niceties already wired up here.
struct QuotesScreen: View {
    @Environment(\.palette) private var palette

    let state: OddlyAppState
    let onBack: () -> Void
    let onShare: () -> Void

    @State private var index: Int
    @State private var favorites: Set<String> = []

    private let quotes = QuoteSeed.shared.all

    init(state: OddlyAppState, onBack: @escaping () -> Void, onShare: @escaping () -> Void) {
        self.state = state
        self.onBack = onBack
        self.onShare = onShare
        let all = QuoteSeed.shared.all
        let today = state.today.epochDays
        _index = State(initialValue: ((today % all.count) + all.count) % all.count)
    }

    var body: some View {
        let quote = quotes[index]
        let isFavorite = favorites.contains(quote.id)

        ZStack {
            palette.background.ignoresSafeArea()
            StarField(starCount: 55, seed: 47).ignoresSafeArea()

            VStack(spacing: 0) {
                OddlyTopBar(title: "", onBack: onBack)

                VStack(spacing: 0) {
                    Text("Cảm hứng mỗi ngày")
                        .font(OddlyFont.headlineSmall)
                        .foregroundStyle(palette.textPrimary)

                    Button {
                        if isFavorite { favorites.remove(quote.id) } else { favorites.insert(quote.id) }
                    } label: {
                        OddlyIconView(
                            .heart,
                            size: 20,
                            tint: isFavorite ? OddlyColors.pink : palette.textTertiary
                        )
                        .frame(width: 44, height: 44)
                        .background(
                            isFavorite ? OddlyColors.pink.opacity(0.18) : palette.surfaceElevated,
                            in: Circle()
                        )
                    }
                    .buttonStyle(PressableStyle())
                    .padding(.top, 20)

                    Spacer(minLength: 0)

                    quoteCard(quote)

                    HStack(spacing: 20) {
                        CircleIconButton(.chevronLeft, diameter: 36, iconSize: 16, tint: palette.textSecondary) {
                            index = (index - 1 + quotes.count) % quotes.count
                        }
                        Text("\(index + 1) / \(quotes.count)")
                            .font(OddlyFont.labelSmall)
                            .tracking(0.5)
                            .foregroundStyle(palette.textTertiary)
                        CircleIconButton(.chevronRight, diameter: 36, iconSize: 16, tint: palette.textSecondary) {
                            index = (index + 1) % quotes.count
                        }
                    }
                    .padding(.top, 20)

                    Spacer(minLength: 0)

                    GradientButton("Chia sẻ", leadingIcon: .share, action: onShare)

                    reminderRow
                        .padding(.top, 20)
                        .padding(.bottom, 28)
                }
                .padding(.horizontal, 24)
            }
        }
        .toolbar(.hidden, for: .navigationBar)
    }

    private func quoteCard(_ quote: Quote) -> some View {
        VStack(alignment: .leading, spacing: 20) {
            Text("“\(quote.text)”")
                .font(OddlyFont.headlineSmall)
                .foregroundStyle(palette.textPrimary)
                .fixedSize(horizontal: false, vertical: true)
            Text("– \(quote.author)")
                .font(OddlyFont.bodyMedium)
                .foregroundStyle(palette.textSecondary)
                .frame(maxWidth: .infinity, alignment: .trailing)
        }
        .padding(28)
        .frame(maxWidth: .infinity, alignment: .leading)
        .background(
            LinearGradient(
                colors: [OddlyColors.purple.opacity(0.16), palette.surfaceElevated],
                startPoint: .top,
                endPoint: .bottom
            ),
            in: RoundedRectangle(cornerRadius: OddlyRadius.extraLarge, style: .continuous)
        )
    }

    private var reminderRow: some View {
        HStack {
            VStack(alignment: .leading, spacing: 0) {
                Text("Lời nhắc")
                    .font(OddlyFont.titleSmall)
                    .foregroundStyle(palette.textPrimary)
                Text("Bật thông báo hằng ngày")
                    .font(OddlyFont.bodySmall)
                    .foregroundStyle(palette.textTertiary)
            }
            .frame(maxWidth: .infinity, alignment: .leading)

            OddlyToggle(isOn: Binding(
                get: { state.settings.reminderEnabled },
                set: { state.settings = state.settings.with(reminderEnabled: $0) }
            ))
        }
        .padding(.horizontal, 16)
        .padding(.vertical, 12)
        .background(
            palette.surfaceElevated,
            in: RoundedRectangle(cornerRadius: OddlyRadius.medium, style: .continuous)
        )
    }
}

File: androidApp/src/main/kotlin/dev/lildua/oddly/ui/screens/quotes/QuotesScreen.kt
package dev.lildua.oddly.ui.screens.quotes

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.statusBarsPadding
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Switch
import androidx.compose.material3.SwitchDefaults
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableIntStateOf
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import dev.lildua.oddly.data.seed.QuoteSeed
import dev.lildua.oddly.ui.components.GradientButton
import dev.lildua.oddly.ui.components.OddlyIcon
import dev.lildua.oddly.ui.components.OddlyTopBar
import dev.lildua.oddly.ui.components.StarField
import dev.lildua.oddly.ui.components.clickableNoRipple
import dev.lildua.oddly.ui.state.OddlyAppState
import dev.lildua.oddly.ui.theme.OddlyColors
import dev.lildua.oddly.ui.theme.LocalStrings
import dev.lildua.oddly.ui.theme.localized
import dev.lildua.oddly.ui.theme.OddlyTheme

/**
 * S14 — one quote a day, from the bundled offline database. Swiping through the
 * archive and favouriting are P1 niceties already wired up here.
 */
@Composable
fun QuotesScreen(
    state: OddlyAppState,
    onBack: () -> Unit,
    onShare: () -> Unit,
) {
    val palette = OddlyTheme.palette
    val strings = LocalStrings.current
    val quotes = QuoteSeed.all
    val todayIndex = remember { ((state.today.toEpochDays() % quotes.size) + quotes.size) % quotes.size }
    var index by remember { mutableIntStateOf(todayIndex) }
    val favorites = remember { mutableStateOf(setOf<String>()) }

    val quote = quotes[index].localized()
    val isFavorite = quote.id in favorites.value

    Box(
        Modifier
            .fillMaxSize()
            .background(MaterialTheme.colorScheme.background),
    ) {
        StarField(Modifier.fillMaxSize(), starCount = 55, seed = 47)

        Column(
            Modifier
                .fillMaxSize()
                .statusBarsPadding(),
        ) {
            OddlyTopBar(title = "", onBack = onBack)

            Column(
                modifier = Modifier
                    .fillMaxSize()
                    .padding(horizontal = 24.dp),
                horizontalAlignment = Alignment.CenterHorizontally,
            ) {
                Text(
                    text = strings.quotesTitle,
                    style = MaterialTheme.typography.headlineSmall,
                    color = palette.textPrimary,
                )

                Spacer(Modifier.height(20.dp))

                Box(
                    modifier = Modifier
                        .size(44.dp)
                        .clip(CircleShape)
                        .background(
                            if (isFavorite) OddlyColors.Pink.copy(alpha = 0.18f)
                            else palette.surfaceElevated,
                        )
                        .clickableNoRipple {
                            favorites.value = if (isFavorite) {
                                favorites.value - quote.id
                            } else {
                                favorites.value + quote.id
                            }
                        },
                    contentAlignment = Alignment.Center,
                ) {
                    OddlyIcon(
                        OddlyIcon.Heart,
                        size = 20.dp,
                        tint = if (isFavorite) OddlyColors.Pink else palette.textTertiary,
                    )
                }

                Spacer(Modifier.weight(1f))

                // Quote card
                Column(
                    modifier = Modifier
                        .fillMaxWidth()
                        .clip(RoundedCornerShape(24.dp))
                        .background(
                            Brush.verticalGradient(
                                listOf(
                                    OddlyColors.Purple.copy(alpha = 0.16f),
                                    palette.surfaceElevated,
                                ),
                            ),
                        )
                        .padding(28.dp),
                ) {
                    Text(
                        text = "“${quote.text}”",
                        style = MaterialTheme.typography.headlineSmall,
                        color = palette.textPrimary,
                    )
                    Spacer(Modifier.height(20.dp))
                    Text(
                        text = "– ${quote.author}",
                        style = MaterialTheme.typography.bodyMedium,
                        color = palette.textSecondary,
                        modifier = Modifier.fillMaxWidth(),
                        textAlign = TextAlign.End,
                    )
                }

                Spacer(Modifier.height(20.dp))

                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.Center,
                    verticalAlignment = Alignment.CenterVertically,
                ) {
                    NavArrow(OddlyIcon.ChevronLeft) {
                        index = (index - 1 + quotes.size) % quotes.size
                    }
                    Spacer(Modifier.size(20.dp))
                    Text(
                        text = "${index + 1} / ${quotes.size}",
                        style = MaterialTheme.typography.labelSmall,
                        color = palette.textTertiary,
                    )
                    Spacer(Modifier.size(20.dp))
                    NavArrow(OddlyIcon.ChevronRight) {
                        index = (index + 1) % quotes.size
                    }
                }

                Spacer(Modifier.weight(1f))

                GradientButton(text = strings.share, onClick = onShare, leadingIcon = OddlyIcon.Share)

                Spacer(Modifier.height(20.dp))

                Row(
                    modifier = Modifier
                        .fillMaxWidth()
                        .clip(RoundedCornerShape(16.dp))
                        .background(palette.surfaceElevated)
                        .padding(horizontal = 16.dp, vertical = 12.dp),
                    verticalAlignment = Alignment.CenterVertically,
                ) {
                    Column(Modifier.weight(1f)) {
                        Text(
                            text = strings.reminderLabel,
                            style = MaterialTheme.typography.titleSmall,
                            color = palette.textPrimary,
                        )
                        Text(
                            text = strings.reminderToggleBody,
                            style = MaterialTheme.typography.bodySmall,
                            color = palette.textTertiary,
                        )
                    }
                    Switch(
                        checked = state.settings.reminderEnabled,
                        onCheckedChange = {
                            state.settings = state.settings.copy(reminderEnabled = it)
                        },
                        colors = SwitchDefaults.colors(
                            checkedThumbColor = MaterialTheme.colorScheme.onPrimary,
                            checkedTrackColor = OddlyColors.Success,
                        ),
                    )
                }

                Spacer(Modifier.height(28.dp))
            }
        }
    }
}

@Composable
private fun NavArrow(icon: OddlyIcon, onClick: () -> Unit) {
    val palette = OddlyTheme.palette
    Box(
        modifier = Modifier
            .size(36.dp)
            .clip(CircleShape)
            .background(palette.surfaceElevated)
            .clickableNoRipple(onClick),
        contentAlignment = Alignment.Center,
    ) {
        OddlyIcon(icon, size = 16.dp, tint = palette.textSecondary)
    }
}
